tests/io/hudi/test_table_read.py

```diff
@@ -0,0 +1,30 @@
+from __future__ import annotations
+
+import pyarrow as pa
+import pytest
+
+import daft
+
+PYARROW_LE_10_0_0 = tuple(int(s) for s in pa.__version__.split(".") if s.isnumeric()) < (10, 0, 0)
+pytestmark = pytest.mark.skipif(PYARROW_LE_10_0_0, reason="hudi only supported if pyarrow >= 10.0.0")
```

Comment:
Can these be removed? Doesn't seem to be used anywhere

Reply:
@jaychia this is actually needed to pass the CI "daft / unit-tests-with-coverage (3.7, py, 6.0.1, ubuntu)" which has lower pyarrow version.

Reply:
As discussed offline, it may be possible to drop `pyhudi`'s pin on Arrow >= 10:

```python
with self.fs.open_input_file(path) as f:
    schema = pq.read_schema(f)
```